lib/error_reporter.h

```diff
@@ -231,6 +231,7 @@ class ErrorReporter {
     /// @return the action to take for the given diagnostic, falling back to the
     /// default action if it wasn't overridden via the command line or a pragma.
     DiagnosticAction getDiagnosticAction(cstring diagnostic, DiagnosticAction defaultAction) {
+        if (defaultAction == DiagnosticAction::Error) return defaultAction;
```

Comment:
Add a comment here that actions for errors can never be overridden. Maybe even return `DiagnosticAction::Error` always?

Reply:
There is a `ingnored` typo in the comment below.

Reply:
> Maybe even return `DiagnosticAction::Error` always?

Do you mean like this:
```
if (defaultAction == DiagnosticAction::Error) return DiagnosticAction::Error;
```
What is the benefit?

Reply:
Well, it would make it explicit and one could not circumvent this invariant by passing in another default action. But if someone goes through the trouble of doing that they may have a reason.